### PR TITLE
Delay creating snapshot

### PR DESCRIPTION
**Pull request by rkondratowicz**
Before creating a snapshot of existing file, check that:
- files was not already deleted from the source container
- sas token was successfully retrieved

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobMover.java [hunk 1]
```diff
@@ -45,27 +41,27 @@ public void moveToRejectedContainer(String blobName, String containerName) {
...
-                new BlobServiceSasSignatureValues(
-                    OffsetDateTime.of(LocalDateTime.now().plus(5, ChronoUnit.MINUTES), ZoneOffset.UTC),
-                    new BlobContainerSasPermission().setReadPermission(true)
-                )
-            );
-        targetBlob
-            .getBlockBlobClient()
-            .copyFromUrl(sourceBlob.getBlockBlobClient().getBlobUrl() + "?" + sasToken);
-
-        logger.info("File successfully uploaded to rejected container. " + loggingContext);
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobMover.java:66]: this has gone from right hand side diff. intentional?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobMover.java:66]: yeah, this and then "File successfully moved to rejected container." just next to it looked weird.

**Comment by aliveni** [comment]
No changes to the tests?

**Comment by aliveni** [comment]
> No changes to the tests?

never mind. code just moved

**Pull request merged**